src/service/customers/CustomersService.ts

```diff
@@ -1,12 +1,71 @@
 import { injectable } from "inversify";
 import Service from "../base-classes/Service";
+import { parse as uuidParse } from "uuid";
 import CustomersDAO from "../../dao/customers/CustomersDAO";
+import PromotionDAO from "../../dao/promotions/PromotionsDAO";
+import RedeemedPromotionDAO from "../../dao/promotions/RedeemedPromotionsDAO";
 import Customer from "../../model/Customer";
 
 @injectable()
 class CustomersService extends Service<Customer> {
-  constructor(protected readonly _customersDAO: CustomersDAO) {
+  constructor(protected readonly _customersDAO: CustomersDAO, protected readonly _promotionsDAO: PromotionDAO, protected readonly _redeemedPromotionsDAO: RedeemedPromotionDAO) {
     super(_customersDAO);
+    this._promotionsDAO = _promotionsDAO;
+    this._redeemedPromotionsDAO = _redeemedPromotionsDAO;
+  }
+
+  async getRelatedPurchases(id: string) {
+    return this._customersDAO.getRelatedPurchases(id);
+  }
+
+  async getRelatedPets(id: string) {
+    return this._customersDAO.getRelatedPets(id);
+  }
+
+  async getRelatedRedeemedPromotions(customerId: string, promotionId: string) {
+    return this._customersDAO.getRelatedRedeemedPromotions(customerId, promotionId);
+  }
+
+  async hasPurchaseOlderThan(customerId: string, date: Date) {
+    const purchases: any = await this.getRelatedPurchases(customerId);
```

Comment:
I know it is against Typescript to use the "any" type as it defeats the purpose. I tried using the objection model as the type however when I access a property such as date, I would get an error saying the property does not exist on the model. I even tried using "&" with an interface but got the same error.